### PR TITLE
Framework: Installation Testing

### PR DESCRIPTION
**Pull request by william76**
@bartlettroscoe here's the branch I'm looking at with the installation testing behavior that we discussed today.  I'll add the configure commands I"ve been using on the cee-compute cluster for my build-install and build-test configurations into comments below.

### CODE REVIEW

**Comment by william76** [comment]
@bartlettroscoe Here's the configuration lines:

### Build-Install 
```bash
cmake \
  -DCMAKE_C_FLAGS:STRING="-Wall -Wempty-body -Wignored-qualifiers -Wmissing-field-initializers -Wsign-compare -Wtype-limits -Wuninitialized -Winit-self -Wstrict-aliasing -Wno-deprecated-declarations -Wno-unused-local-typedefs -Wno-unused-variable -Wno-shadow -g -fdiagnostics-color=auto" \
  -DCMAKE_CXX_FLAGS:STRING="-Wall -Wempty-body -Wignored-qualifiers -Wmissing-field-initializers -Wsign-compare -Wtype-limits -Wuninitialized -Winit-self -Wstrict-aliasing -Wno-deprecated-declarations -Wno-unused-local-typedefs -Wno-unused-variable -Wno-shadow -g -fdiagnostics-color=auto" \
  -DCMAKE_CXX_COMPILER=/projects/sems/install/rhel6-x86_64/sems/compiler/gcc/7.2.0/openmpi/1.10.1/bin/mpic++ \
  -DCMAKE_C_COMPILER=/projects/sems/install/rhel6-x86_64/sems/compiler/gcc/7.2.0/openmpi/1.10.1/bin/mpicc \
  -DCOVERAGE_COMMAND=/projects/sems/install/rhel6-x86_64/sems/compiler/gcc/7.2.0/base/bin/gcov \
  -DMPI_BIN_DIR:FILEPATH=/projects/sems/install/rhel6-x86_64/sems/compiler/gcc/7.2.0/openmpi/1.10.1/bin \
  -DMPI_CXX_COMPILER=/projects/sems/install/rhel6-x86_64/sems/compiler/gcc/7.2.0/openmpi/1.10.1/bin/mpic++ \
  -DMPI_C_COMPILER=/projects/sems/install/rhel6-x86_64/sems/compiler/gcc/7.2.0/openmpi/1.10.1/bin/mpicc \
  -DMPI_EXEC=/projects/sems/install/rhel6-x86_64/sems/compiler/gcc/7.2.0/openmpi/1.10.1/bin/mpiexec \
  -DTrilinos_ENABLE_Fortran:BOOL=OFF \
  -DCMAKE_INSTALL_PREFIX:FILEPATH=/scratch/wcmclen/dev/trilinos-dev/build/fmwk-bkw-compat-build/../fmwk-bkw-compat-install \
  -DTrilinos_ENABLE_TESTS:BOOL=OFF \
  -DTrilinos_ENABLE_Gtest:BOOL=ON \
  -DTrilinos_ENABLE_Kokkos:BOOL=ON \
  -DKokkos_ENABLE_TESTS:BOOL=OFF \
  -DTrilinos_ENABLE_MueLu:BOOL=ON \
  -DTrilinos_ENABLE_Teuchos:BOOL=ON \
  -DTrilinos_ENABLE_Tpetra:BOOL=ON \
  -DTrilinos_ENABLE_TpetraCore:BOOL=ON \
  -DTrilinos_ENABLE_Xpetra:BOOL=ON \
  ${TRILINOS_SOURCE_DIR:?}
```

### Build-Test
```bash
cmake \
  -DCMAKE_C_FLAGS:STRING="-Wall -Wempty-body -Wignored-qualifiers -Wmissing-field-initializers -Wsign-compare -Wtype-limits -Wuninitialized -Winit-self -Wstrict-aliasing -Wno-deprecated-declarations -Wno-unused-local-typedefs -Wno-unused-variable -Wno-shadow -g -fdiagnostics-color=auto" \
  -DCMAKE_CXX_FLAGS:STRING="-Wall -Wempty-body -Wignored-qualifiers -Wmissing-field-initializers -Wsign-compare -Wtype-limits -Wuninitialized -Winit-self -Wstrict-aliasing -Wno-deprecated-declarations -Wno-unused-local-typedefs -Wno-unused-variable -Wno-shadow -g -fdiagnostics-color=auto" \
  -DCMAKE_CXX_COMPILER=/projects/sems/install/rhel6-x86_64/sems/compiler/gcc/7.2.0/openmpi/1.10.1/bin/mpic++ \
  -DCMAKE_C_COMPILER=/projects/sems/install/rhel6-x86_64/sems/compiler/gcc/7.2.0/openmpi/1.10.1/bin/mpicc \
  -DCOVERAGE_COMMAND=/projects/sems/install/rhel6-x86_64/sems/compiler/gcc/7.2.0/base/bin/gcov \
  -DMPI_BIN_DIR:FILEPATH=/projects/sems/install/rhel6-x86_64/sems/compiler/gcc/7.2.0/openmpi/1.10.1/bin \
  -DMPI_CXX_COMPILER=/projects/sems/install/rhel6-x86_64/sems/compiler/gcc/7.2.0/openmpi/1.10.1/bin/mpic++ \
  -DMPI_C_COMPILER=/projects/sems/install/rhel6-x86_64/sems/compiler/gcc/7.2.0/openmpi/1.10.1/bin/mpicc \
  -DMPI_EXEC=/projects/sems/install/rhel6-x86_64/sems/compiler/gcc/7.2.0/openmpi/1.10.1/bin/mpiexec \
  -DTrilinos_ENABLE_Fortran:BOOL=OFF \
  -DCMAKE_INSTALL_PREFIX:FILEPATH=/scratch/wcmclen/dev/trilinos-dev/build/fmwk-bkw-compat-test/../fmwk-bkw-compat-install \
  -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON \
  -DTrilinos_ENABLE_TESTS:BOOL=ON \
  -DTrilinos_INSTALLATION_DIR:FILEPATH=/scratch/wcmclen/dev/trilinos-dev/build/fmwk-bkw-compat-test/../fmwk-bkw-compat-install \
  -DTrilinos_ENABLE_Gtest:BOOL=ON \
  -DGtest_ENABLE_TESTS:BOOL=OFF \
  -DTrilinos_ENABLE_Kokkos:BOOL=ON \
  -DKokkos_ENABLE_TESTS:BOOL=OFF \
  -DKokkosCore_ENABLE_TESTS:BOOL=OFF \
  -DKokkosContainers_ENABLE_TESTS:BOOL=OFF \
  -DKokkosAlgorithms_ENABLE_TESTS:BOOL=OFF \
  -DTrilinos_ENABLE_MueLu:BOOL=ON \
  -DTrilinos_ENABLE_Teuchos:BOOL=ON \
  -DTrilinos_ENABLE_Tpetra:BOOL=ON \
  -DTrilinos_ENABLE_TpetraCore:BOOL=ON \
  -DTrilinos_ENABLE_Xpetra:BOOL=ON \
  ${TRILINOS_SOURCE_DIR:?}
```
When we spoke on this we also noted that I got configure errors when I try to enable `-DTrilinos_ENABLE_INSTALLATION_TESTING:BOOL=ON` on my configure line for the test job.  It was that odd error citing that some library was test only but not test only...

**Comment by william76** [comment]
@jhux2 I wouldn't burn too much energy reviewing this PR... it's a WIP and could change alot before anything in here might get merged in.  I didn't tag the MueLu team intentionally since this PR is mostly just me and Ross looking at what's going on in the CMake files :p

File: packages/muelu/test/unit_tests/CMakeLists.txt [hunk 1]
```diff
@@ -359,3 +359,18 @@ IF (${PACKAGE_NAME}_ENABLE_Epetra)
...
+# -----------------------------------------
+#SET(HEADERS "")
+#APPEND_SET(HEADERS
+#    MueLu_Test_ETI.hpp
+#    MueLu_TestHelpers.hpp
+#    MueLu_TestHelpers_Common.hpp
+#    )
+#TRIBITS_INSTALL_HEADERS(HEADERS)
+#MESSAGE(STATUS ">>> MueLU TEST/UNIT_TEST/HEADERS: ${HEADERS}")
+
```
[the remark this answers is not in the document]
Comment (william76) [reply packages/muelu/test/unit_tests/CMakeLists.txt]: WIP

**Comment by bartlettroscoe** [comment]
@william76, wash this closed on purpose?  Is the effort to add installation and backward compatibility testing for Trilinos dead?

**Comment by william76** [comment]
no, it's just pretty old and I was cleaning things out.  Installation testing has been interrupted so many times that when I can get back to it it's probably going to be about like starting new again.  I'm pretty sure getting muelu to work will still be broken because it doesn't handle testing headers correctly when installing but for an initial effort to work out what jenkins and the new python test driver scripts look like I'll be maybe just turning on teuchos only or something like that to get the overall mechanics of it together.

**Comment by bartlettroscoe** [comment]
Related to:
* SEPW-213

**Pull request closed without merging**